pkg/clean/clean.go

```diff
@@ -52,9 +51,9 @@ func Clean(ctx context.Context, cf *config.NSXOperatorConfig, logg *logr.Logger,
 	// Clean may be called many times, only init once
 	once.Do(func() {
 		if logg != nil {
-			logf.SetLogger(*logg)
+			log = *logg
 		} else {
-			logf.SetLogger(logger.ZapLogger(debug, logLevel))
+			log = logger.ZapLogger(debug, logLevel)
 		}
 	})
```

Comment:
Actually, I think the premise of this is incorrect. Suppose multiple clusters invoke `Clean`, they should each operate with independent loggers, right?

Instead of a package-level:

```go
var log = logger.Log
```

What if you just used:

```
if logg == nil {
    logg = logger.ZapLogger(debug, logLevel)
}

logg.Info...
```

and pass `logg` down to each consuming function that needs it as an added required argument? You can then remove the package global as it's no longer necessary.

The package scope is pretty small, and this means you have a contained logger per instance of `Clean`. This also helps in case retries offer contextual data.

Reply:
it's correct.
so using passing log inside clean package.
but controller runtime still needs SetLogger.
it complained while I removed the SetLogger.
[controller-runtime] log.SetLogger(...) was never called; logs will not be displayed.